### PR TITLE
Video 7918 add speakers list

### PR DESCRIPTION
**Pull request by olipyskoty**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [VIDEO-7918](https://issues.corp.twilio.com/browse/VIDEO-79180)

### Description

This PR adds the list of speakers to the Participants panel. If the user is the Host, they will see menu icons next to each speaker where they will have the ability to control the state of the speaker (i.e., mute them, remove them, etc,). Note, these menu items are static for now, and the functionality will be implemented in a future PR. All non-host users will only see the names of the speakers.

**Host View:**
Screen Shot 2021-12-30 at 1 12 06 PM

**Host View Menu Open** (ignore blue box around menu, not sure why my computer added that when I took a screenshot)
Screen Shot 2021-12-30 at 1 12 23 PM

**Non-host View**
Screen Shot 2021-12-30 at 1 16 36 PM

### Before review
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [x] Verified locally with all effected platforms
* [x] Manually sanity tested running locally
* [x] Included screenshot as PR comment (if needed)
* [x] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

File: apps/web/src/components/ParticipantWindow/ParticipantWindow.tsx [hunk 1]
```diff
@@ -54,6 +55,7 @@ export default function ParticipantWindow() {
   return (
     <aside className={clsx(classes.participantWindowContainer, { [classes.hide]: !appState.isParticipantWindowOpen })}>
```
Comment (timmydoza) [inline apps/web/src/components/ParticipantWindow/ParticipantWindow.tsx:56]: I think we'll need to add some CSS to make this list scrollable when it gets long, but we can probably handle that in the Viewers List PR, since the Speakers List and Viewers List will both need to be able to scroll within the ParticipantWindow.

**Review by timmydoza — CHANGES_REQUESTED** [review]
Looks good! Since some SpeakersList functionality is coming in a later PR, we should merge this one into a feature branch instead of main.

File: apps/web/src/components/SpeakersList/SpeakersList.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    justifyContent: 'space-between',
+    width: '100%',
+    alignItems: 'center',
+  },
+});
+
+export default function SpeakersList() {
+  const { host, speakers } = useSpeakersMap();
+  const { room } = useVideoContext();
+  const localParticipant = room!.localParticipant;
```
Comment (timmydoza) [inline apps/web/src/components/SpeakersList/SpeakersList.tsx:31]: Just had a thought - the ParticipantWindow should be available to Viewers too right? In addition to the Host and Speakers? 

Feels like it should be easy to do, but I think we'd need to handle the case for Viewers where `room` is `null`.

Comment (olipyskoty) [reply apps/web/src/components/SpeakersList/SpeakersList.tsx:31]: Looks like all that was needed was some optional chaining! I also added the participant window to the `<PlayerMenuBar />` component. I decided to add a new simple `toggleParticipantWindow()` function since the `ToggleParticipantWindowButton` component contained a lot more functionality than what the Viewer needs.
Screen Shot 2022-01-05 at 1 14 43 PM

File: apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx [hunk 3]
```diff
@@ -71,6 +82,14 @@ export default function PlayerMenuBar({ roomName, disconnect }: { roomName?: str
           </Grid>
         </Grid>
 
+        <Grid item>
+          <Grid container justifyContent="center">
+            <Button onClick={() => toggleParticipantWindow()} startIcon={<ParticipantIcon />}>
+              Participants
+            </Button>
+          </Grid>
+        </Grid>
```
Comment (timmydoza) [inline apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx:91]: I'm not sure that this button needs to be in its own grid. I think we can put it in the same grid as the Hand Raise button.
```
        <Grid item>
          <Grid container justifyContent="center">
            <Button onClick={handleRaiseHand} startIcon={isHandRaised ? <LowerHandIcon /> : <RaiseHandIcon />}>
              {isHandRaised ? 'Lower Hand' : 'Raise Hand'}
            </Button>
            <Button onClick={() => toggleParticipantWindow()} startIcon={<ParticipantIcon />}>
              Participants
            </Button>
          </Grid>
        </Grid>
```

File: apps/web/src/components/Player/Player.tsx [hunk 4]
```diff
@@ -30,6 +37,7 @@ function Player() {
   const classes = useStyles();
   const videoElRef = useRef<HTMLVideoElement>(null!);
   const { player, disconnect } = usePlayerContext();
+  const { isChatWindowOpen } = useChatContext();
```
Comment (timmydoza) [inline apps/web/src/components/Player/Player.tsx:40]: I don't think the chat stuff needs to be here without a ChatWindow for the viewers. We should probably add it for the viewers soon, but that can be another PR.

**Review by timmydoza — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx [hunk 5]
```diff
@@ -37,6 +39,7 @@ const useStyles = makeStyles((theme: Theme) =>
 export default function PlayerMenuBar({ roomName, disconnect }: { roomName?: string; disconnect: () => void }) {
   const classes = useStyles();
   const { appState, appDispatch } = useAppState();
+  const { setIsChatWindowOpen } = useChatContext();
```
Comment (timmydoza) [inline apps/web/src/components/Player/PlayerMenuBar/PlayerMenuBar.tsx:42]: Sorry - should have mentioned it but I think chat stuff here can go too.

**Review by timmydoza — CHANGES_REQUESTED** [review]

**Review by timmydoza — APPROVED** [review]
